packages/extension/src/background/preAuthorizationMessaging.ts

```diff
@@ -28,7 +28,7 @@ export const handlePreAuthorizationMessage: HandleMessage<
         openUi()
         return
       }
-      const authorized = await getPreAuthorized(msg.data.host)
```

Comment:
If we are already connected to `devnet` and now we want to switch to `testnet`, the `getPreauthorized` returns a preauthorization because we only check the host, so the window does not pop up, and then we will still connect to `devnet`.